upgrading.md

```diff
@@ -0,0 +1,170 @@
+# Upgrading Philanthropy Data Commons dependencies
+
+This document guides one through upgrades for the back-end services at
+https://api.philanthropydatacommons.org and
+https://auth.philanthropydatacommons.org.
+
+The front-end at https://pilot.philanthropydatacommons.org has its own process.
+
+This applies to the non-PDC-web-container services. As of this writing that is
+the auth, database, and reverse-proxy services. The rest is either in the PDC
+web service, such as node dependencies managed in the `service` project, or on
+the host, such as cert-bot.
+
+See the summary in `deploying.md` regarding how software gets deployed. The same
+explanation regarding compose.yml[1] applies.
+
+In general, try it on a local development environment first, then make sure it
+succeeds in the test environment, then finally do it in production.
+
+[1]: https://github.com/PhilanthropyDataCommons/deploy/blob/main/compose.yml
+
+## Upgrade postgresql (minor version)
+
+Caution: do not use "latest". That won't work the way we want. Do not use a
+non-specific version such as 14-debian-11 or 14.8.0. Use only specific versions.
+The purpose of using bitnami images is reproducibility of issues by virtue of
+reproducibility of images via immutable tag names.
+
+0. Clone or pull the repo[2].
+1. In your local clone of the repo[2], open the `compose.yml` file.
+2. Look at the current version of the `database` image under "database: image:".
+3. Visit dockerhub[3] and search for "bitnami postgresql"[4], or go straight to
+the bitnami/postgresql[5] images and click "Tags". Or straight there[6].
+4. Find the most recent and highest version of the major version of postgresql.
+For example, if using postgresql 14, it might be 14.8.0-debian-11-r14.
+5. Replace the version in the `compose.yml` file and save the file.
+6. Test the pull locally with that compose.yml:
+  - `docker compose -f compose.yml pull` (Some systems: `docker-compose`).
+If the pull succeeded, continue, otherwise re-check the previous couple steps.
+7. Create a branch, add the compose file, commit the changes, push the changes:
+  - `git checkout -b upgrade-to-postgresql-image-[version]`
+  - `git add compose.yml`
+  - `git commit` (Use a message indicating the new version)
+  - `git push origin upgrade-to-postgresql-image-[version]`
+8. Create a pull request to merge that branch (follow the link git prints).
```

Comment:
I think you don't need to say "(follow the link git prints)" here.

In general, you can make some assumptions about the audience of a document like this -- you have to, really.  One of those assumptions is that the person has basic familiarity with using GitHub (or other GitHub-like forge platforms like GitLab) as a developer.  I think that's a safe assumption.  We don't need to teach them how PRs work.